DSoundManagerクラスのリファクタリングと修正

Moto0116 opened the pull request:
**修正内容**
 - 関数の仮引数名修正
 - CスタイルのキャストはC++スタイルに
 - 変数名の修正
 - 参照をポインタに修正

issue #160

taishi-matsumura commented on src/KingdomCraft/KingdomCraft/Library/DSoundManager/DSoundManager.cpp:
```diff
@@ -33,154 +34,163 @@ bool DSoundManager::Init()
-	DWORD fmsize = formatChunk.cksize;
-	DWORD size = mmioRead(hMmio, (HPSTR)&_wFmt, fmsize);
-	if (size != fmsize)
+	if (!OpenWave(_pFileName, &WaveFormat, &pWaveData, &WaveSize))
```
`OpenWave()`の中でファイルをcloseしちゃうんですね。
だったら`OpenWave()`じゃなくて`ReadWave()`とかの方が良いのでは？
(「closeしなくていいの？」と思ってしまいました)

Moto0116 replied:
すみません、その通りです。
修正しときました

taishi-matsumura reviewed (approved):
関数名に対する指摘です

motolz committed:
[issue #160] DSoundManagerクラスのリファクタリング

- コメントの付与
- 仮引数名の修正とconstの付与
- enumをクラス内部に修正
- エラー時の処理を修正
- 参照を使ってたのをポインタに修正
- コンストラクタで初期化するメンバにはconst付与
- 変数名の修正
- private関数にコメント付与
- ReleaseSound関数でNULLチェックを行うようにした
- OpenWave関数の名前を修正
